internal/query/execution.go

```diff
@@ -0,0 +1,197 @@
+package query
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/api/call"
+	"github.com/zitadel/zitadel/internal/database"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/query/projection"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+var (
+	executionTable = table{
+		name:          projection.ExecutionTable,
+		instanceIDCol: projection.ExecutionInstanceIDCol,
+	}
+	ExecutionColumnID = Column{
+		name:  projection.ExecutionIDCol,
+		table: executionTable,
+	}
+	ExecutionColumnCreationDate = Column{
+		name:  projection.ExecutionCreationDateCol,
+		table: executionTable,
+	}
+	ExecutionColumnChangeDate = Column{
+		name:  projection.ExecutionChangeDateCol,
+		table: executionTable,
+	}
+	ExecutionColumnResourceOwner = Column{
+		name:  projection.ExecutionResourceOwnerCol,
+		table: executionTable,
+	}
+	ExecutionColumnInstanceID = Column{
+		name:  projection.ExecutionInstanceIDCol,
+		table: executionTable,
+	}
+	ExecutionColumnSequence = Column{
+		name:  projection.ExecutionSequenceCol,
+		table: executionTable,
+	}
+	ExecutionColumnTargets = Column{
+		name:  projection.ExecutionTargetsCol,
+		table: executionTable,
+	}
+	ExecutionColumnIncludes = Column{
+		name:  projection.ExecutionIncludesCol,
+		table: executionTable,
+	}
+)
+
+type Executions struct {
+	SearchResponse
+	Executions []*Execution
+}
+
+func (e *Executions) SetState(s *State) {
+	e.State = s
+}
+
+type Execution struct {
+	ID string
+	*domain.ObjectDetails
```

Comment:
IMO i'd either have an explicit field or remove the pointer
current solution is prone for nil pointer panics